src/utils/fromJS.js

```diff
@@ -7,6 +7,21 @@
  */
 
 const Immutable = require("immutable");
+const isFunction = require("lodash/isFunction");
+
+// hasOwnProperty is defensive because it is possible that the
+// object that we're creating a map for has a `hasOwnProperty` field
+function hasOwnProperty(value, key) {
```

Comment:
We don't. I just made the PR. The test shows the issue though. Objects we want to create a map for can have the field `hasOwnProperty`